ksmt-yices/src/main/kotlin/org/ksmt/solver/yices/KYicesExprInternalizer.kt

```diff
@@ -0,0 +1,666 @@
+        transform(value) { value: YicesTerm -> Terms.bvSignExtend(value, extensionSize) }
+    }
+
+    override fun transform(expr: KBvZeroExtensionExpr): KExpr<KBvSort> = with(expr) {
+        transform(value) { value: YicesTerm -> Terms.bvZeroExtend(value, extensionSize) }
+    }
+
+    override fun transform(expr: KBvRepeatExpr): KExpr<KBvSort> = with(expr) {
+        transform(value) { value: YicesTerm -> Terms.bvRepeat(value, repeatNumber) }
+    }
+
+    override fun <T : KBvSort> transform(expr: KBvShiftLeftExpr<T>): KExpr<T> = with(expr) {
+        transform(arg0, arg1, Terms::bvShl)
+    }
+
+    override fun <T : KBvSort> transform(expr: KBvLogicalShiftRightExpr<T>): KExpr<T> = with(expr) {
+        transform(arg0, arg1, Terms::bvLshr)
+    }
+
+    override fun <T : KBvSort> transform(expr: KBvArithShiftRightExpr<T>): KExpr<T> = with(expr) {
+        transform(arg0, arg1, Terms::bvAshr)
+    }
+
+    override fun <T : KBvSort> transform(expr: KBvRotateLeftExpr<T>): KExpr<T> = with(ctx) {
+        with(expr) {
+            transform(arg0, arg1) { arg0: YicesTerm, arg1: YicesTerm ->
+                val size = expr.sort.sizeBits
+                val bvSize = Terms.bvConst(size.toInt(), size.toLong())
+                val rotationNumber = Terms.bvRem(arg1, bvSize)
+
+                val left = Terms.bvShl(arg0, rotationNumber)
+                val right = Terms.bvLshr(arg0, Terms.bvSub(bvSize, rotationNumber))
+
+                Terms.bvAdd(left, right)
+            }
+        }
+    }
+
+    override fun <T : KBvSort> transform(expr: KBvRotateLeftIndexedExpr<T>): KExpr<T> = with(expr) {
+        transform(value) { value: YicesTerm -> Terms.bvRotateLeft(value, rotationNumber) }
+    }
+
+    override fun <T : KBvSort> transform(expr: KBvRotateRightExpr<T>): KExpr<T> = with(ctx) {
+        with(expr) {
+            transform(arg0, arg1) { arg0: YicesTerm, arg1: YicesTerm ->
+                val size = expr.sort.sizeBits
+                val bvSize = Terms.bvConst(size.toInt(), size.toLong())
+                val rotationNumber = Terms.bvRem(arg1, bvSize)
+
+                val left = Terms.bvShl(arg0, Terms.bvSub(bvSize, rotationNumber))
+                val right = Terms.bvLshr(arg0, rotationNumber)
+
+                Terms.bvAdd(left, right)
+            }
+        }
+    }
+
+    override fun <T : KBvSort> transform(expr: KBvRotateRightIndexedExpr<T>): KExpr<T> = with(expr) {
+        transform(value) { value: YicesTerm -> Terms.bvRotateRight(value, rotationNumber) }
+    }
+
+    override fun transform(expr: KBv2IntExpr): KExpr<KIntSort> = with(ctx) {
+        with(expr) {
+            transform(value) { value: YicesTerm ->
+                val size = expr.value.sort.sizeBits.toInt()
+                val twoConst = Terms.intConst(2)
+
+                val args = (0 until size - 1).map {
+                    Terms.ifThenElse(
+                        Terms.bvExtractBit(value, it),
+                        Terms.power(twoConst, it),
```

Comment:
It's better to compute an actual integer constant instead of creating symbolic operations.